test/modules/core/lib/widget-manager.spec.js

```diff
@@ -0,0 +1,362 @@
+/* global document */
+import test from 'tape-promise/tape';
+
+import {WidgetManager, Widget} from '@deck.gl/core/lib/widget-manager';
+import {WebMercatorViewport} from '@deck.gl/core';
+
+class TestWidget extends Widget {
```

Comment:
Have you considered covering a test widget with an overridden constructor, and/or covering the constructed, but not added, lifecycle stage where many members are expected to be undefined?